frontend/src/components/QuickAdd.tsx

```diff
@@ -20,6 +20,9 @@ export const QuickAdd = () => {
       .then((res) => {
         if (res.ok) {
           return res.json();
+        } else if (res.status === 401) {
+          // Redirect to login page
+          window.location.href = "/";
```

Comment:
Looks good to me. But better to use the function useNavigate, that is an evolution of the useHistory, see the Login page for an example.

Reply:
It adds lot of extra LoC at the moment, but I added it in a new commit